frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+    message = "drats, condition is false";
+}
+message === "nice, condition is true";
+
+// you could bring that down to something like this using a ternary
+message = condition ? "nice, condition is true" : "drats, condition is false";
+message === "nice, condition is true";
+```
+
+### Arrays
+Arrays let you bundle multiple values into one data structure.
+```js
+// array literals are delimited by square brackets ([])
+[0, 1, 2, 3]
+// elements don't have to be the same type
+let arr = ["value zero", "value one", 2, false];
+// elements are accessed using the subscript ([]) operator
+// arrays are zero-indexed, meaning they start counting their elements at zero
+// yields "value zero"
+arr[0]:
+// yields "value one"
+arr[1];
+// yields 2
+arr[2];
+// you can change the contents at an index by reassigning with =
+// after this next line, arr will look like ["value zero", "value one", 2, 3]
+arr[3] = 3;
+// you can even store other arrays inside of an array!
+let nested = [["array 0 element 0"], ["array 1 element 0"], ["array 2 element 0"]];
+// you access these values using chained subscripts. the first subscript yields
+// the element stored at that index (which is another array) and the second one
+// subscripts into that yielded array. this yields "array 1 element 0".
+nested[1][0];
+// you can get an array's length using the `.length` property (more on
+// properties in the classes section). this yields 4.
+arr.length;
+```
+
+### Loops
+Loops let you repeat a block of code while a condition is true. They come in three primary varieties: `while`, `for`, and `for-of`.
+```js
+// while loops continue so long as the condition in their parenthesis (())
+// yields true
+let howMany = 0;
+let message = "We looped ";
+while (howMany < 5) {
+    message += howMany + " ";
+    howMany += 1;
+}
+message += "times";
+message === "We looped 0 1 2 3 4 times";
+
+// for loops let you declare and use a variable scoped to the loop, usually for
+// iteration. they generally look like `for (initialization; condition; step)`.
+message = "We looped ";
+for (let howMuch = 0; howMuch < 5; howMuch += 1) {
+    message += howMuch + " ";
+}
+message += "times";
+message === "We looped 0 1 2 3 4 times";
+
+// for loops can be thought of as just a while loop in a trench coat. the
+// previous example could have instead been written equivalently as:
+{
+    let howMuch = 0;
+    while (howMuch < 5) {
+        message += howMuch + " ";
+        howMuch += 1;
+    }
+    message += "times";
+}
+
+// for-of loops are analogous to for-each loops in other languages. they
+// let you loop through collections of things.
+message = "We've seen elements ";
+let array = [0, 1, 2, 3, 4, 5];
+// trying to change the variable you bind each element to will just change
+// the variable, not the array, so we usually const-declare them for clarity
+for (const element of array) {
+    message += element + " "
+}
+// message === "We've seen elements 0 1 2 3 4 5 "
+```
+
+### Functions
```

Comment:
Should we talk about default parameters, parameter destructuring, varargs, generators, and/or closures?

Reply:
I would argue for varargs and default params, but I can't speak to any of those other functionalities. Perhaps Avery or Nivedha have a stance on them.